File: mobile_client/app/src/main/java/wpam/mobile_client/protocol/BlindsStatusCommand.java
package wpam.mobile_client.protocol;import java.util.ArrayList;

public class BlindsStatusCommand extends Command
	{
		public BlindsStatusCommand()
		{
			super(10, CommandType.BLINDS_STATUS);
		}
		public final void close()
		{
			super.close();
		}

		@Override
		public ArrayList<Integer> getFrameBytes()
		{
			initializeDataSize();

			ArrayList<Integer> frame = getHeader();
			frame.add(commandType_.getValue());

			return frame;
		}
		@Override
		public String getName()
		{
			return new String("BlindsStatusCommand");
		}

		@Override
		public void accept(CommandVisitor visitor)
		{
			visitor.visit(this);
		}

		@Override
		protected void initializeDataSize()
		{
			setDataSize(1);
		}
	}

File: server/src/protocol/BlindsStatusCommand.cpp
#include "BlindsStatusCommand.h"
#include "CommandVisitor.h"

#include <utility/BytesConverter.h>

using namespace std;
using namespace utility;
using namespace communication;

BlindsStatusCommand::BlindsStatusCommand()
    : Command(10, CommandType::BLINDS_STATUS)
{}

BlindsStatusCommand::~BlindsStatusCommand()
{}

vector<uint8_t> BlindsStatusCommand::getFrameBytes()
{
    initializeDataSize();

    vector<uint8_t > frame = getHeader();
    frame.push_back(static_cast<uint8_t>(commandType_));

    return frame;
}

string BlindsStatusCommand::getName()
{
    return string("BlindsStatusCommand");
}

void BlindsStatusCommand::accept(CommandVisitor &visitor)
{
    visitor.visit(*this);
}

void BlindsStatusCommand::initializeDataSize()
{
    uint16_t dataSize = sizeof(commandType_);

    setDataSize(dataSize);
}
